pandas/tests/tseries/test_frequencies.py

```diff
@@ -7,6 +7,10 @@
 from pandas import (Index, DatetimeIndex, Timestamp, Series,
                     date_range, period_range)
 
+from pandas._libs.tslibs.frequencies import (_period_code_map, get_rule_month,
+                                             _period_str_to_code,
+                                             _INVALID_FREQ_ERROR)
```

Comment:
are these methods marked non-private (_period_st_to_code) used outside of libfrequencies? if so, de-privatize (and cpdef them). same with INVALID_FREQ_ERR

Reply:
Mostly just in tests.  If OK I'm going to save the remaining de-privatization and test_libfreqs for a follow-up.  Too many active PRs and this one is ready for its AARP membership.

Reply:
well I would rather get it right here. as we progress more and more the PR's by-definition get more fine grained and I care much more about the details. when just moving stuff, sure that doesn't matter.

Reply:
_INVALID_FREQ_ERROR is used once in the non-cython frequencies.  The others are used in tests but otherwise not outside libfreqs